lib/ansible/modules/system/systemd.py

```diff
@@ -448,8 +448,8 @@ def main():
                         if rc != 0:
                             module.fail_json(msg="Unable to %s service %s: %s" % (action, unit, err))
             else:
-                # this should not happen?
-                module.fail_json(msg="Service is in unknown state", status=result['status'])
+                # this can happen when we're running in a chroot, in which case systemd supports only enable/disable operations
```

Comment:
i would not change the default ... either have a flag to change the behaviour or 'chroot' detection

Reply:
Are we sure about that? systemctl already performs chroot detection, and obviously until now this case is leading to a fatal error, in a code path that the original author said "should not happen" anyway, so there is no real backward compatibility requirement to fulfil here. With these preconditions, I wonder whether we can't apply this simple patch that automatically upgrades a relatively obscure code path that's just failing for now to a better behaviour by default, rather than accumulating cruft by introducing an extra flag that users must set.

Reply:
if you see the `?` on the note, i added it as i still had to research what corner cases could this affect. If systemctl already detects the chroot, does it have a way of telling us this is the case?

Reply:
No, the exit code isn't specific. The check in systemctl is this one:

https://github.com/systemd/systemd/blob/df0ff127758809a45105893772de76082d12a26d/src/basic/virt.c#L572

(where files_same compares device IDs and inode numbers). I guess we could replicate that in Python easily enough. Your plan would be to perform this check and continue on only if we detect we're running in a chroot, and otherwise fail just like before? I'd concur with that and I can implement this. I just don't want another module parameter if it can be avoided. :)

Reply:
I've added a corresponding change. Please review. :)